src/overlays/actors/ovl_En_Ice_Hono/z_en_ice_hono.c

```diff
@@ -21,37 +32,345 @@ const ActorInit En_Ice_Hono_InitVars = {
+            EnIceHono_InitDroppedFlame(&this->actor, globalCtx);
+            break;
+        case 1:
+        case 2:
+            EnIceHono_InitSmallFlame(&this->actor, globalCtx);
+            break;
+    }
+
+    if ((this->actor.params == -1) || (this->actor.params == 0)) {
+        Lights_PointNoGlowSetInfo(&this->lightInfo, this->actor.posRot.pos.x,
+                                        (s16)this->actor.posRot.pos.y + 0x0A, this->actor.posRot.pos.z, 0x9B, 0xD2,
+                                        0xFF, 0);
+        this->lightNode = LightContext_InsertLight(globalCtx, &globalCtx->lightCtx, &this->lightInfo);
+        this->unk154 = Math_Rand_ZeroOne() * 65535.5f;
+        this->unk156 = Math_Rand_ZeroOne() * 65535.5f;
+        // Translates to: "(ice flame)"
+        osSyncPrintf("(ice 炎)(arg_data 0x%04x)\n", this->actor.params);
+    }
+}
+
+void EnIceHono_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnIceHono* this = THIS;
+
+    if ((this->actor.params == -1) || (this->actor.params == 0)) {
+        LightContext_RemoveLight(globalCtx, &globalCtx->lightCtx, this->lightNode);
+        Collider_DestroyCylinder(globalCtx, &this->collider);
+    }
+}
+
+u32 EnIceHono_LinkCloseAndFacing(EnIceHono* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (this->actor.xzDistFromLink < 60.0f) {
+        Vec3f tempPos;
+        tempPos.x = Math_Sins(this->actor.yawTowardsLink + 0x8000) * 40.0f + player->actor.posRot.pos.x;
+        tempPos.y = player->actor.posRot.pos.y;
+        tempPos.z = Math_Coss(this->actor.yawTowardsLink + 0x8000) * 40.0f + player->actor.posRot.pos.z;
+        if (EnIceHono_SquareDist(&tempPos, &this->actor.posRot.pos) <= 1600.0f) {
+            return 1;
+        }
+    }
+    return 0;
+}
+
+void EnIceHono_SetupActionCapturableFlame(EnIceHono* this) {
+    this->actionFunc = EnIceHono_CapturableFlame;
+    this->alpha = 0xFF;
+    this->actor.shape.unk_08 = -1000.0f;
+}
+
+void EnIceHono_CapturableFlame(EnIceHono* this, GlobalContext* globalCtx) {
+    if (Actor_HasParent(&this->actor, globalCtx))
+        this->actor.parent = 0;
+    else if (EnIceHono_LinkCloseAndFacing(this, globalCtx))
+        func_8002F434(&this->actor, globalCtx, 0x7E, 60.0f, 100.0f);
+
+    if (this->actor.xzDistFromLink < 200.0f) {
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    }
+    func_8002F8F0(&this->actor, 0x20A3);
+}
+
+void EnIceHono_SetupActionDroppedFlame(EnIceHono* this) {
+    this->actionFunc = EnIceHono_DropFlame;
+    this->timer = 0xC8;
+    this->alpha = 0xFF;
+}
+
+void EnIceHono_DropFlame(EnIceHono* this, GlobalContext* globalCtx) {
+    u32 bgFlag = this->actor.bgCheckFlags & 1;
+
+    Math_ApproxF(&this->actor.scale.x, 0.0017f, 0.00008f);
+    this->actor.scale.z = this->actor.scale.x;
+    Math_ApproxF(&this->actor.scale.y, 0.0017f, 0.00008f);
+
+    if (bgFlag != 0) {
+        s32 i;
+        for (i = 0; i != 0x10000; i += 0x2000) {
```

Comment:
does `i < 0x10000` work?
also can you replace this with `i` going from 0 to 16 (exclusive) and replace `i` usage in the loop by `i*0x2000` ?